.github/workflows/run_tests_and_checks.yml

```diff
@@ -42,4 +42,11 @@ jobs:
     - name: Unused imports and variables check
       if: success() || failure()
       run: |
-        python -m autoflake --remove-all-unused-imports -r . --exclude venv,archived_code -c --quiet
\ No newline at end of file
+        python -m autoflake --remove-all-unused-imports -r . --exclude venv,archived_code -c --quiet
+
+    - name: Run unittest
```

Comment:
btw can we add an `python3 -m pip freeze` somewhere here? This should make easier investigating which package version change broke tests.